src/styles/base.scss

```diff
@@ -0,0 +1,59 @@
+body {
+  margin: 0;
+  --column-count: 1;
+}
+
+
+
+.main {
+  padding-top: 50px;
+  padding-inline: 40px;
+  display: grid;
+  grid-template-columns: repeat(var(--column-count), $card-width);
+  row-gap: 48px;
+  column-gap: 46px;
+  justify-content: center;
+}
+
+@media (min-width: 488px) {
+  .main {
+    --column-count: 2;
+  }
+}
+
+@media (min-width: 768px) {
+  .main {
+    --column-count: 3;
+  }
+}
+
+@media (min-width: 1024px) {
+  .main {
+    --column-count: 4;
+  }
+}
+
+.product {
+  margin-top: 1px;
+  margin-left: 1px;
+  width: $card-width;
+  height: $card-height;
```

Comment:
The variable `$card-height` is used here, but it is not defined in this file. Ensure that `$card-height` is defined somewhere in your SCSS files before it is used.